### ISSUE TITLE
Cause: error=13, Permission denied

### ISSUE THREAD

**Issue by sunildhiman90** [android/nowinandroid#322]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is there a StackOverflow question about this issue?

- [X] I have searched StackOverflow

### What happened?

While importing sample in my new mac m1 pro, getting this error, NOt able to import the project.

* Exception is:
org.gradle.process.internal.ExecException: A problem occurred starting process 'command 'tools/setup.sh''
	at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:241)
	at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:218)
	at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:370)
	at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:87)
	at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
Caused by: net.rubygrapefruit.platform.NativeException: Could not start 'tools/setup.sh'
	at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27)
	at net.rubygrapefruit.platform.internal.WrapperProcessLauncher.start(WrapperProcessLauncher.java:36)
	at org.gradle.process.internal.ExecHandleRunner.startProcess(ExecHandleRunner.java:98)
	at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:71)
	... 3 more
Caused by: java.io.IOException: Cannot run program "tools/setup.sh" (in directory "/Users/sunil/AndroidStudioProjects/nowinandroid"): error=13, Permission denied
	at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
	... 6 more
Caused by: java.io.IOException: error=13, Permission denied
	... 7 more

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sunildhiman90** [comment]
But i was cloning with Dolphin itself in first time, still i am getting this error. I tried this 2 times, but in both turns, it was the same error , no luck. I dont know what this issue is related to?

**Comment by georgiscoolman** [comment]
Are you doing "New -> Project from Version Control" ->  URL "https://github.com/android/nowinandroid.git" ?

**Comment by sunildhiman90** [comment]
I think i tried with import sample option. But i will try with this option of direct url as well.

**Comment by AviranAbady** [comment]
Fails to build in Windows as well, unable to run the setup.sh bash script - `Caused by: java.io.IOException: CreateProcess error=193, %1 is not a valid Win32 application`

```
* Exception is:
org.gradle.process.internal.ExecException: A problem occurred starting process 'command 'tools/setup.sh''
	at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:241)
	at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:218)
	at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:370)
	at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:87)
	at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
Caused by: net.rubygrapefruit.platform.NativeException: Could not start 'tools/setup.sh'
	at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27)
	at net.rubygrapefruit.platform.internal.WindowsProcessLauncher.start(WindowsProcessLauncher.java:22)
	at net.rubygrapefruit.platform.internal.WrapperProcessLauncher.start(WrapperProcessLauncher.java:36)
	at org.gradle.process.internal.ExecHandleRunner.startProcess(ExecHandleRunner.java:98)
	at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:71)
	... 3 more
Caused by: java.io.IOException: Cannot run program "tools/setup.sh" (in directory "D:\astudio\nowinandroid"): CreateProcess error=193, %1 is not a valid Win32 application
	at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
	... 7 more
Caused by: java.io.IOException: CreateProcess error=193, %1 is not a valid Win32 application
	... 8 more




```

**Comment by AviranAbady** [comment]
As a temporary workaround, commented out the following in `settings.gradle.kts`

```
//if (!hooksInstalled) {
//    exec {
//        commandLine("tools/setup.sh")
//        workingDir = rootProject.projectDir
//    }
//}
```

**Comment by dturner** [comment]
Thanks for reporting this. It should have been fixed in #335.
